examples/www-image/main/jpg-render.c

```diff
@@ -0,0 +1,518 @@
+#include "esp_heap_caps.h"
+#include "esp_log.h"
+#include "esp_timer.h"
+#include "esp_types.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "freertos/event_groups.h"
+#include "esp_sleep.h"
+// WiFi related
+#include "esp_wifi.h"
+#include "esp_event.h"
+#include "esp_log.h"
+#include "nvs_flash.h"
+#include "lwip/err.h"
+#include "lwip/sys.h"
+// HTTP Client
+#include "esp_netif.h"
+#include "esp_tls.h"
+#include "esp_http_client.h"
+#include "esp_crt_bundle.h"
+
+// JPG decoder
+#if ESP_IDF_VERSION_MAJOR >= 4 // IDF 4+
+  #include "esp32/rom/tjpgd.h"
+#else // ESP32 Before IDF 4.0
+  #include "rom/tjpgd.h"
+#endif
+
+#include "esp_task_wdt.h"
+#include <stdio.h>
+#include <string.h>
+#include <math.h> // round + pow
+
+#include "epd_driver.h"
+#include "epd_highlevel.h"
+EpdiyHighlevelState hl;
+
+// WiFi configuration
+#define ESP_WIFI_SSID     "WLAN-724300"
+#define ESP_WIFI_PASSWORD "50238634630558382093"
+
+
+// Image URL and jpg settings. Make sure to update WIDTH/HEIGHT if using loremflickr
+// Note: Only HTTP protocol supported (Check README to use SSL secure URLs)
+#define IMG_URL "https://loremflickr.com/960/540"
```

Comment:
I'd add something like this:
`
#define STR_HELPER(x) #x
#define STR(x) STR_HELPER(x)

#define IMG_URL ("https://loremflickr.com/"STR(EPD_WIDTH)"/"STR(EPD_HEIGHT))
`
So the images are in the right resolution. One less thing to change for the user.

Reply:
Great advice, thanks!